src/mongo_types.ts

```diff
@@ -541,3 +525,60 @@ export type NestedPathsOfType<TSchema, Type> = KeysOfAType<
   },
   Type
 >;
+
+/**
+ * @public
+ * @experimental
+ */
+export type DotNotationFilter<TSchema> =
+  | Partial<TSchema>
+  | ({
+      [Property in Join<NestedPaths<WithId<TSchema>, []>, '.'>]?: Condition<
+        PropertyType<WithId<TSchema>, Property>
+      >;
+    } & RootFilterOperators<WithId<TSchema>>);
+
+/**
+ * @public
+ * @experimental
+ */
+export type DotNotationUpdateFilter<TSchema> = {
+  $currentDate?: OnlyFieldsOfType<
+    TSchema,
+    Date | Timestamp,
+    true | { $type: 'date' | 'timestamp' }
+  >;
+  $inc?: OnlyFieldsOfType<TSchema, NumericType | undefined>;
+  $min?: DotNotationMatchKeysAndValues<TSchema>;
+  $max?: DotNotationMatchKeysAndValues<TSchema>;
+  $mul?: OnlyFieldsOfType<TSchema, NumericType | undefined>;
+  $rename?: Record<string, string>;
+  $set?: DotNotationMatchKeysAndValues<TSchema>;
+  $setOnInsert?: DotNotationMatchKeysAndValues<TSchema>;
+  $unset?: OnlyFieldsOfType<TSchema, any, '' | true | 1>;
+  $addToSet?: SetFields<TSchema>;
+  $pop?: OnlyFieldsOfType<TSchema, ReadonlyArray<any>, 1 | -1>;
+  $pull?: PullOperator<TSchema>;
+  $push?: PushOperator<TSchema>;
+  $pullAll?: PullAllOperator<TSchema>;
+  $bit?: OnlyFieldsOfType<
+    TSchema,
+    NumericType | undefined,
+    { and: IntegerType } | { or: IntegerType } | { xor: IntegerType }
+  >;
+} & Document;
+
+/** @public */
```

Comment:
Do we want this to be public as well or should we mark internal?

Reply:
The 2 should be public but I was referring specifically to `DotNotationMatchKeysAndValues` - does it need to be public because the other 2 are?

Reply:
Oh, sorry.  I misread your message.

Yeah, if we mark this as internal, we get api-extractor warnings.

Reply:
Ah ok - thanks for clarifying.